static/js/compose.js

```diff
@@ -565,8 +565,20 @@ exports.validate = function () {
 };
 
 exports.handle_keydown = function (event) {
-    var code = event.keyCode || event.which;
     var textarea = $("#compose-textarea");
+
+    // Set the rtl class if the text has an rtl direction, remove it otherwise
+    var text = textarea.val();
+    if (text.startsWith('```quote')) {
+        text = text.substr(8);
+    }
```

Comment:
I was worried that this implementation may cause performance problems when typing a very long message; every time a user types a character, we're processing the entire text of the message (via `textarea.val()`, which means effectively we're doing quadratic work.  

But I did some testing with the Chrome profiler, and it appears that even with a 10K character message (maximum size) there's no noticable impact.  So I think we're fine on that front.